### PR TITLE
New Package: NetworkManager-fortisslvpn-1.3.90

### PR DESCRIPTION
**Pull request by StellarOrbit**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+lib32disabled=yes
+_nm_openfortivpn_homedir="/var/empty"
+
+do_configure() {
+	./configure --prefix=/usr \
+	--sysconfdir=/etc \
+	--localstatedir=/var \
+	--libexecdir=/usr/lib/networkmanager \
+	--disable-static \
+	--with-pppd-plugin-dir=/usr/lib/pppd/$_pppver
```
Comment (abenson) [inline srcpkgs/NetworkManager-fortisslvpn/template:26]: `_pppver` is never defined.

Also, you can probably drop all of this and replace it with

    configure_args="--libexecdir=/usr/lib/networkmanager --with-pppd-plugin-dir=/usr/lib/pppd/$_pppver"

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+depends="openfortivpn"
+short_desc="NetworkManager VPN plugin for OpenFortiVPN"
+maintainer="Twilighttony <starrbeartony14@zoho.com>"
+license="GPL-2.0-or-later"
+homepage="https://wiki.gnome.org/Projects/NetworkManager"
+distfiles="${GNOME_SITE}/${pkgname}/${version%.*}/${pkgname}-${version}.tar.xz"
+checksum=884a7432bb38ae9c942e094f23d0a3feb504432153c3d2869ab71a4dd244d0fb
+system_accounts="_nm_openfortivpn"
+lib32disabled=yes
+_nm_openfortivpn_homedir="/var/empty"
```
Comment (abenson) [inline srcpkgs/NetworkManager-fortisslvpn/template:18]: Maybe give it its own directory?  Does it store any files?

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Template file for 'NetworkManager-openvpn'
```
Comment (cinerea0) [inline srcpkgs/NetworkManager-fortisslvpn/template:1]: This should be the same as the package name.

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	--sysconfdir=/etc \
+	--localstatedir=/var \
+	--libexecdir=/usr/lib/networkmanager \
+	--disable-static \
+	--with-pppd-plugin-dir=/usr/lib/pppd/$_pppver
+  make
+}
+
+post_extract() {
+	sed -i 's/\(#define NM_OPENFORTIVPN_\(USER\|GROUP\)\).*/\1 "_nm_openfortivpn"/' \
```
Comment (cinerea0) [inline srcpkgs/NetworkManager-fortisslvpn/template:31]: You should probably use [vsed](https://github.com/void-linux/void-packages/blob/master/Manual.md#global-functions) for this.

File: srcpkgs/NetworkManager-fortisslvpn/template (hunk 3 above)
Comment (Twilighttony) [reply srcpkgs/NetworkManager-fortisslvpn/template:1]: I’m surprised I didn’t catch that earlier. I’ll change that now.

File: srcpkgs/NetworkManager-fortisslvpn/template (hunk 1 above)
Comment (Twilighttony) [reply srcpkgs/NetworkManager-fortisslvpn/template]: Added and committed.

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+license="GPL-2.0-or-later"
+homepage="https://wiki.gnome.org/Projects/NetworkManager"
+distfiles="${GNOME_SITE}/${pkgname}/${version%.*}/${pkgname}-${version}.tar.xz"
+checksum=884a7432bb38ae9c942e094f23d0a3feb504432153c3d2869ab71a4dd244d0fb
+system_accounts="_nm_openfortivpn"
+lib32disabled=yes
+_nm_openfortivpn_homedir="/var/empty"
+_pppver=2.4.9
+
+configure_args="--libexecdir=/usr/lib/networkmanager --with-pppd-plugin-dir=/usr/lib/pppd/$_pppver"
```
Comment (cinerea0) [inline srcpkgs/NetworkManager-fortisslvpn/template:21]: You should probably move this to the "configure_args" line above.

File: srcpkgs/NetworkManager-fortisslvpn/template (hunk 2 above)
Comment (abenson) [reply srcpkgs/NetworkManager-fortisslvpn/template:18]: Nevermind, I see that tracks with the other plugins.

File: srcpkgs/NetworkManager-fortisslvpn/template (hunk 5 above)
Comment (Twilighttony) [reply srcpkgs/NetworkManager-fortisslvpn/template]: I have fixed this now.

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+license="GPL-2.0-or-later"
+homepage="https://wiki.gnome.org/Projects/NetworkManager"
+distfiles="${GNOME_SITE}/${pkgname}/${version%.*}/${pkgname}-${version}.tar.xz"
+checksum=884a7432bb38ae9c942e094f23d0a3feb504432153c3d2869ab71a4dd244d0fb
+system_accounts="_nm_openfortivpn"
+lib32disabled=yes
+_nm_openfortivpn_homedir="/var/empty"
+_pppver=2.4.9
+
+configure_args="--disable-static --libexecdir=/usr/lib/networkmanager --with-pppd-plugin-dir=/usr/lib/pppd/$_pppver"
```
Comment (abenson) [inline srcpkgs/NetworkManager-fortisslvpn/template:20]: This should go after `build_style`

Check with `xlint` from `xtools`

Comment (Twilighttony) [reply srcpkgs/NetworkManager-fortisslvpn/template]: Done

**Comment by the-maldridge** [comment]
Needs rebase.

**Comment by Twilighttony** [comment]
> Needs rebase.

Done.

**Comment by classabbyamp** [comment]
related issue that can be closed if this is merged: #20462

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'NetworkManager-fortisslvpn'
+pkgname=NetworkManager-fortisslvpn
+version=1.3.90
+revision=1
+build_style=gnu-configure
+_pppver=2.4.9
+configure_args="--disable-static --libexecdir=/usr/lib/networkmanager --with-pppd-plugin-dir=/usr/lib/pppd/$_pppver"
```
Comment (classabbyamp) [inline srcpkgs/NetworkManager-fortisslvpn/template:7]:
```suggestion
configure_args="--disable-static --with-pppd-plugin-dir=/usr/lib/pppd/$_pppver"
```
this is not correct, other networkmanager vpn clients use `/usr/libexec`

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+hostmakedepends="automake gettext-devel glib-devel intltool libtool pkg-config"
+makedepends="ppp-devel gtk+3-devel libglib-devel libnma-devel libsecret-devel"
+depends="openfortivpn"
+short_desc="NetworkManager VPN plugin for OpenFortiVPN"
+maintainer="Twilighttony <starrbeartony14@zoho.com>"
+license="GPL-2.0-or-later"
+homepage="https://wiki.gnome.org/Projects/NetworkManager"
+distfiles="${GNOME_SITE}/${pkgname}/${version%.*}/${pkgname}-${version}.tar.xz"
```
Comment (classabbyamp) [inline srcpkgs/NetworkManager-fortisslvpn/template:15]:
```suggestion
distfiles="${GNOME_SITE}/NetworkManager-fortisslvpn/${version%.*}/NetworkManager-fortisslvpn-${version}.tar.xz"
```

**Review by classabbyamp — COMMENTED** [review]
I'm having trouble with this, I get this error when trying to activate a fortissl vpn connection via networkmanager:

```
2023-02-03T18:24:03.86897 daemon.info: Feb  3 13:24:03 NetworkManager[924]: <info>  [1675448643.8688] audit: op="connection-activate" uuid="..." name="Foo" pid=3677 uid=1000 result="fail" reason="The VPN service 'org.freedesktop.NetworkManager.fortisslvpn' was not installed."
```

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'NetworkManager-fortisslvpn'
+pkgname=NetworkManager-fortisslvpn
+version=1.3.90
+revision=1
+build_style=gnu-configure
+_pppver=2.4.9
+configure_args="--disable-static --libexecdir=/usr/lib/networkmanager --with-pppd-plugin-dir=/usr/lib/pppd/$_pppver"
+hostmakedepends="automake gettext-devel glib-devel intltool libtool pkg-config"
+makedepends="ppp-devel gtk+3-devel libglib-devel libnma-devel libsecret-devel"
+depends="openfortivpn"
```
Comment (slimjimsoftware) [inline srcpkgs/NetworkManager-fortisslvpn/template:10]:
```suggestion
depends="openfortivpn ppp"
```

**Comment by slimjimsoftware** [comment]
> I'm having trouble with this, I get this error when trying to activate a fortissl vpn connection via networkmanager:
> 
> ```
> 2023-02-03T18:24:03.86897 daemon.info: Feb  3 13:24:03 NetworkManager[924]: <info>  [1675448643.8688] audit: op="connection-activate" uuid="..." name="Foo" pid=3677 uid=1000 result="fail" reason="The VPN service 'org.freedesktop.NetworkManager.fortisslvpn' was not installed."
> ```

This depends on ppp and you need to reboot after installing, I can confirm after this change and a reboot it works on aarch64

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'NetworkManager-fortisslvpn'
+pkgname=NetworkManager-fortisslvpn
+version=1.3.90
```
Comment (Eloitor) [inline srcpkgs/NetworkManager-fortisslvpn/template:3]: New version is 1.4.0

File: srcpkgs/NetworkManager-fortisslvpn/template [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+build_style=gnu-configure
+_pppver=2.4.9
+configure_args="--disable-static --libexecdir=/usr/lib/networkmanager --with-pppd-plugin-dir=/usr/lib/pppd/$_pppver"
+hostmakedepends="automake gettext-devel glib-devel intltool libtool pkg-config"
+makedepends="ppp-devel gtk+3-devel libglib-devel libnma-devel libsecret-devel"
+depends="openfortivpn"
+short_desc="NetworkManager VPN plugin for OpenFortiVPN"
+maintainer="Twilighttony <starrbeartony14@zoho.com>"
+license="GPL-2.0-or-later"
+homepage="https://wiki.gnome.org/Projects/NetworkManager"
```
Comment (Eloitor) [inline srcpkgs/NetworkManager-fortisslvpn/template:14]: Maybe you could add the changelog

https://gitlab.gnome.org/GNOME/NetworkManager-fortisslvpn/-/raw/main/NEWS

File: srcpkgs/NetworkManager-fortisslvpn/template (hunk 9 above)
Comment (Eloitor) [reply srcpkgs/NetworkManager-fortisslvpn/template:10]: Is this required? I thought that `makedepends="ppp-devel` already makes the package depend on `ppp`...

**Comment by classabbyamp** [comment]
author does not appear to be using github anymore

**Pull request closed without merging**